pkg/errors/wrap.go

```diff
@@ -0,0 +1,39 @@
+package errors
+
+import "reflect"
+
+// Unwrap returns the result of calling the Unwrap method on err, if err's
+// type contains an Unwrap method returning error.
+// Otherwise, Unwrap returns nil.
+func Unwrap(err error) error {
+	u, ok := err.(interface {
+		Unwrap() error
+	})
+	if !ok {
+		return nil
+	}
+	return u.Unwrap()
+}
+
+// Is for seeing if an error includes the target error
+func Is(err, target error) bool {
```

Comment:
This Is a generic package Is. 

We use this as errors.Is(err, typeoferr).  This will call the included Is function that you have highlighted if it exists for the Error.  Not all errors are required to have Is implemented so this provides that logic.